python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,498 @@
+        result.source_footprints["detector"] = result.exposure.info.getDetector().getId()
+        result.background = background
+        self._apply_photo_calib(result.exposure, result.source_footprints, photo_calib)
+
+        # TODO: do we need this, or can we drop LocalWcs/LocalPhotoCalib?
+        self.post_calculations.run(result.source_footprints, result.exposure)
+        result.source = result.source_footprints.asAstropy()
+
+        return result
+
+    def _find_sources(self, exposure, background, id_generator):
+        """Detect and measure sources on the exposure.
+
+        Parameters
+        ----------
+        exposures : `lsst.afw.image.Exposure`
+            Exposure to detect and measure sources on; must have a valid PSF.
+        background : `lsst.afw.math.BackgroundList`
+            Background that was fit to the exposure during detection;
+            modified in-place during subsequent detection.
+        id_generator : `lsst.meas.base.IdGenerator`
+            Object that generates source IDs and provides random seeds.
+
+        Returns
+        -------
+        source
+            Catalog that was detected and measured on the exposure.
+        """
+        table = afwTable.SourceTable.make(self.source_schema.schema, id_generator.make_table_id_factory())
+
+        detections = self.detection.run(table=table, exposure=exposure, background=background)
+        source = detections.sources
+
+        self.sky_sources.run(exposure.mask, id_generator.catalog_id, source)
+
+        self.deblend.run(exposure=exposure, sources=source)
+        # The deblender may not produce a contiguous catalog; ensure
+        # contiguity for subsequent tasks.
+        if not source.isContiguous():
+            source = source.copy(deep=True)
+
+        self.measurement.run(source, exposure)
+        self.normalized_calibration_flux.run(exposure=exposure, catalog=source)
+        self.apply_aperture_correction.run(source, exposure.apCorrMap)
+        self.catalog_calculation.run(source)
+        self.set_primary_flags.run(source)
+
+        return source
+
+    def _apply_photo_calib(self, exposure, source_footprints, photo_calib):
+        """Photometrically calibrate the exposure and catalog with the
+        supplied PhotoCalib, and set the exposure's PhotoCalib to 1.
+
+        Parameters
+        ----------
+        exposures : `lsst.afw.image.Exposure`
+            Exposure to calibrate and set PhotoCalib on; Modified in place.
+        source_footprints : `lsst.afw.table.SourceCatalog`
+            Catalog to calibrate.
+        photo_calib : `lsst.afw.image.PhotoCalib`
+            Photometric calibration to apply.
+        calibrated_stars : `lsst.afw.table.SourceCatalog`
+            Star catalog with flux/magnitude columns computed from the
+            supplied PhotoCalib.
+        """
+        source_footprints = photo_calib.calibrateCatalog(source_footprints)
```

Comment:
I think this will do the spatially varying calibration but we've already taken that out.

Reply:
Right, I had to add the `initial_photo_calib` from `calibrateImage`, which I've now donw.